### ISSUE TITLE
No network when using --master-count and existing VPC

### ISSUE THREAD

**Issue by guitmz** [kubernetes/kops#3780]
- Kops: `1.7.1`
- Kubernetes `1.7.9`
- `Calico` as network

To reproduce, create a cluster in an existing VPC in a region where you only have 2 AZs (like Singapore) like this:

```
export NAME=cluster.com
export KOPS_STATE_STORE=s3://my-bucket
export VPC_ID=vpc-whatever
export NETWORK_CIDR=10.64.0.0/16
export DNS_ZONE=my-dns-zone

kops create cluster \
    --api-loadbalancer-type internal \
    --name $NAME \
    --cloud aws \
    --authorization AlwaysAllow \
    --ssh-access ${NETWORK_CIDR} \
    --ssh-public-key my-key.pub \
    --network-cidr ${NETWORK_CIDR} \
    --topology private \
    --dns private \
    --dns-zone ${DNS_ZONE} \
    --associate-public-ip=false \
    --networking calico \
    --node-count 2 \
    --master-count 3 \
    --zones ap-southeast-1a,ap-southeast-1b \
    --master-zones ap-southeast-1a,ap-southeast-1b \
    --node-size c4.2xlarge \
    --node-volume-size 250 \
    --master-size c4.xlarge \
    --master-volume-size 100 \
    --vpc ${VPC_ID} \
    --kubernetes-version 1.7.9
```

The expected is to have the cluster up and running like I have in other regions using the same commands as above (everything is the same on other clusters and regions except `--master-count`, which had to be included here because I need a cluster in `ap-southeast-1`).

- Kubelet logs states the following in all nodes (masters and regular nodes): `Unable to update cni config: No networks found in /etc/cni/net.d/`
- Calico images are not even pulled from registry.
- Note that this does work when I let kops create the VPC.

If there is anything else I may include here, please let me know.

Thanks

**Comment by chrislovecnm** [comment]
You cannot launch 2 instances of etcd. Either run a single master or run three masters in the same az.

**Comment by guitmz** [comment]
I'm sorry but that still doesn't work, please check the command below:

```
export NAME=cluster.com
export KOPS_STATE_STORE=s3://my-bucket
export VPC_ID=vpc-xxxxxxx
export NETWORK_CIDR=10.64.0.0/16
export DNS_ZONE=my-dns-zone

kops create cluster \
    --api-loadbalancer-type internal \
    --name $NAME \
    --cloud aws \
    --authorization AlwaysAllow \
    --ssh-access ${NETWORK_CIDR} \
    --ssh-public-key my-ssh-key.pub \
    --network-cidr ${NETWORK_CIDR} \
    --topology private \
    --dns private \
    --dns-zone ${DNS_ZONE} \
    --associate-public-ip=false \
    --networking calico \
    --node-count 2 \
    --master-count 3 \
    --zones ap-southeast-1a,ap-southeast-1b \
    --master-zones ap-southeast-1a \
    --node-size c4.2xlarge \
    --node-volume-size 250 \
    --master-size c4.xlarge \
    --master-volume-size 100 \
    --vpc ${VPC_ID} \
    --kubernetes-version 1.7.9
```

Additionally, when I let `kops` create the VPC for me it works even with the command on my first post.

**Comment by guitmz** [comment]
I'm comparing a machine in europe that has everything working and `/etc/sysconfig/kubelet` is the same but the contents of `/opt/cni/bin` change:

Europe machine (all working):
```
admin@ip-10-0-10-137:/opt/cni/bin$ ls -lah
total 71M
drwxr-xr-x 2 root root 4.0K Oct 27 10:44 .
drwxr-xr-x 3 root root 4.0K Oct 27 10:42 ..
-rwxr-xr-x 1 root root 3.9M Oct 27 10:42 bridge
-rwxr-xr-x 1 root root  28M Oct 27 10:44 calico
-rwxr-xr-x 1 root root  27M Oct 27 10:44 calico-ipam
-rwxr-xr-x 1 root root 2.8M Oct 27 10:42 flannel
-rwxr-xr-x 1 root root 3.0M Oct 27 10:42 host-local
-rwxr-xr-x 1 root root 3.1M Oct 27 10:42 loopback
-rwxr-xr-x 1 root root 3.9M Oct 27 10:42 ptp
```

Asia machine (calico not present):
```
admin@ip-10-64-60-253:/opt/cni/bin$ ls -lah
total 17M
drwxr-xr-x 2 root root 4.0K Nov  7 09:33 .
drwxr-xr-x 3 root root 4.0K Nov  7 09:33 ..
-rwxr-xr-x 1 root root 3.9M Nov  7 09:33 bridge
-rwxr-xr-x 1 root root 2.8M Nov  7 09:33 flannel
-rwxr-xr-x 1 root root 3.0M Nov  7 09:33 host-local
-rwxr-xr-x 1 root root 3.1M Nov  7 09:33 loopback
-rwxr-xr-x 1 root root 3.9M Nov  7 09:33 ptp
```

Very strange.. I tried running `nodeup` manually on Asia machine but no luck.

**Comment by kenden** [comment]
We got the same issue after upgrading our test cluster from 1.7.2 to 1.7.10 with kops 1.7.1

**Comment by guitmz** [comment]
@kenden i tested again it works fine with kops 1.8 but 1.7.1 is a no go for this scenario

**Comment by chrislovecnm** [comment]
/cc @caseydavenport workaround?

@kenden can you try the beta?  Also, we upgraded calico in kops 1.8

**Comment by caseydavenport** [comment]
@chrislovecnm I don't see a clear diagnosis of what the root cause is here, so not sure what to workaround :)

From the sound of it, there are no CNI network configs showing up on the nodes. This would be a result of Calico not getting scheduled onto those nodes (since Calico installs that config).

Can we identify why Calico isn't getting scheduled? That would get us closer to the root of the problem.

**Comment by guitmz** [comment]
I don't have the nodes anymore but I reproduced the scenario 5 out of 5 times I created the cluster with the settings of my first comment. I tried looking where it failed but I lacked the time to go on with that, unfortunately

**Comment by kenden** [comment]
I tested upgrading a 1.7.2 cluster to 1.7.10 with kops 1.8.0-beta.2, and had no problem this time.

**Comment by justinsb** [comment]
It sounds like this might be resolved in 1.8.0, so I'm going to punt to the 1.8.1 milestone and we can review then?

**Comment by Raffo** [comment]
@justinsb can you link to a PR or commit where you think this is fixed? I am experiencing similar issues but with a different setup and I'm trying to go a bit deeper with that problematic, any help would be beneficial.
